### ISSUE TITLE
Equality and hashcode for (Geo)Position and (Geo)Box classes

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#89]
See also #12 and #58

Add at least to `Position`, `GeoPosition`, `Box` and `GeoBox?`:
* equals operator
* hashCode
* equals2D
* equals3D

**Comment by navispatial** [comment]
Check the new [hash](https://api.dart.dev/stable/2.14.1/dart-core/Object/hash.html) method of the `Object` class (since Dart 2.14).

Sample from official docs:
```dart
class SomeObject {
  final Object a, b, c;
  SomeObject(this.a, this.b, this.c);
  bool operator=(Object other) =>
      other is SomeObject && a == other.a && b == other.b && c == other.c;
  int get hashCode => Object.hash(a, b, c);
}
```

**Comment by navispatial** [comment]
For example in `Position`:

```dart
  @override
  bool operator ==(Object other) =>
      other is Position &&
      x == other.x &&
      y == other.y &&
      z == other.z &&
      m == other.m;

  @override
  int get hashCode => Object.hash(x, y, z, m);
```

**Comment by navispatial** [comment]
Also for `Point` from geocore (that implements `Position`):

```dart
  @override
  bool operator ==(Object other) =>
      other is Point &&
      isGeographic == other.isGeographic &&
      x == other.x &&
      y == other.y &&
      optZ == other.optZ &&
      optM == other.optM;

  @override
  int get hashCode => Object.hash(x, y, optZ, optM);
```

Removing `EquatableMixin` from concrete sub classes of `Point` as it no more needed.

**Comment by navispatial** [comment]
Implemented in pre-release 0.1.0-a.1 of beta 0.1.0 of geobase.
